components/eam/src/physics/cam/zm_conv.F90

```diff
@@ -796,53 +795,36 @@ subroutine zm_convr(lchnk   ,ncol    , &
       dsubcld(i) = 0._r8
    end do
 
-   if( cam_physpkg_is('cam3')) then
+   ! Evaluate Tparcel, qs(Tparcel), buoyancy, CAPE, lcl, lel, parcel launch level at index maxi()=hmax
+   ! - call #1, iclosure=.true.   standard calculation using state of current step
+   ! - call #2, iclosure=.false.  use state from previous step and launch level from call #1
+   ! DCAPE is the difference in CAPE between the two calls using the same launch level
 
-      !  For cam3 physics package, call non-dilute
+   iclosure = .true.
+   call buoyan_dilute(lchnk   ,ncol    ,                             &
+                      q       ,t       ,p       ,z       ,pf       , &
+                      tp      ,qstp    ,tl      ,rl      ,cape     , &
+                      pblt    ,lcl     ,lel     ,lon     ,maxi     , &
+                      rgas    ,grav    ,cpres   ,msg               , &
+                      tpert   ,iclosure                            )
```

Comment:
I find it useful to keep info about intents (in, out, inouts) from a readability perspective. It allows us to find out which quantities are being modified and reason about whether they should be modified or not. If we want to avoid verbosity, we can collect all the intent-ins at the beginning, then inouts, and then intent-outs. This way, we will be adding comments in only three places:
```
call sub( !ins
....
!inouts
....
!outs
....)
```
The majority of the calls may have just ins and outs. You can add this info any way you like.

Reply:
Thanks. I see both perspectives here. I agree that it needs to be maintained (we were enforcing this during the review process in the MAM4 refactor). It may just be a personal preference, but I like to see the "intents" while reading the code to quickly assess whether it makes sense for this call to change these variables or not (without the need to search for the subroutine interface, which might be in another file). It also helps while debugging to immediately know if the variable we are tracing is being modified by a subroutine call or not. That said, I will leave it up to you, I am okay either way.